src/prerequisites/widgets.py

```diff
@@ -0,0 +1,36 @@
+from functools import reduce
+
+from django.db.models import Q
+
+from queryset_sequence import QuerySetSequence
+
+from utilities.widgets import ContentObjectSelect2Widget
+
+
+class CustomContentObjectSelect2Widget(ContentObjectSelect2Widget):
```

Comment:
Can you add a docstring to this class explaining why a customization of `ContentObjectSelect2Widget` is required?  What exactly is begin added or overridded for this widget? Thanks!

Reply:
The problem is that `QuerySetSequenc.filter(field_name=...)` will fail if some models in combined queryset doesn't have the same `field_name`, for example `Quest` and `Badge` models has field `name`, but not `title`. `Course` model has field `title`, but not field `name`.

```python
qs = QuerySetSequenc(Quest.objects.all(), Badge.objects.all(), Course.objects.all())
# this will raise an exception
qs.filter(name__icontains=...)
# this will also fail due to exception
qs.filter(title__icontains=...)
# ... and this will not work
qs.filter(title__icontains=..., name__icontains=...)
```

Reply:
I wrote `ContentObjectSelect2Widget` to match the API of `ModelSelect2Widget` class, see https://github.com/codingjoe/django-select2/blob/main/django_select2/forms.py#L539

Reply:
As solution we can make it work like this,
```python
ContentObjectSelect2Widget(
  search_fields=[
    {Course: ["title__contains"]},
    "name__icontains",
  ]
)
```
and by changing `get_search_fields` and `filter_queryset` methods.

Let me know what do you think.

Reply:
DAL had this same issue, which we solved by forcing the models to implement a method to return the search field name.
```
class IsAPrereqMixin:
...
    @staticmethod
    def dal_autocomplete_search_fields():
        """ 
        For Django Autocomplete Light, should return a field on the model, 
        or something more complicated as described in
        https://django-autocomplete-light.readthedocs.io/en/master/gfk.html#form-example

        override this static method in the class to choose different search fields
        """
        return "name"
```

Models implementng IsAPrereqMixin that did not have a `name` field would override this method and return 'title', for example.

Reply:
solved in https://github.com/bytedeck/bytedeck/pull/1276